addons/website_sale/models/sale_order.py

```diff
@@ -135,17 +142,21 @@ def sale_get_order(self, cr, uid, ids, force_create=False, code=None, update_pri
         # create so if needed
         if not sale_order_id and (force_create or code):
             # TODO cache partner_id session
-            partner = self.pool['res.users'].browse(cr, SUPERUSER_ID, uid, context=context).partner_id
+            user_obj = self.pool['res.users']
+            partner = user_obj.browse(cr, SUPERUSER_ID, uid, context=context).partner_id
+            affiliate_id = request.session.get('affiliate_id')
+            salesperson_id = affiliate_id if user_obj.search(cr, SUPERUSER_ID, [('id', '=', affiliate_id)], context=context) else False
 
             for w in self.browse(cr, uid, ids):
                 values = {
-                    'user_id': w.user_id.id,
+                    'user_id': salesperson_id if salesperson_id else w.user_id.id,
                     'partner_id': partner.id,
                     'pricelist_id': partner.property_product_pricelist.id,
-                    'team_id': self.pool.get('ir.model.data').get_object_reference(cr, uid, 'website', 'salesteam_website_sales')[1],
+                    'team_id': w.team_id.id,
                 }
                 sale_order_id = sale_order_obj.create(cr, SUPERUSER_ID, values, context=context)
                 values = sale_order_obj.onchange_partner_id(cr, SUPERUSER_ID, [], partner.id, context=context)['value']
+                values.update({'user_id': salesperson_id if salesperson_id else w.user_id.id})
```

Comment:
because ,  everytime onchange_partner_id call it return different user id then  we set in create 
so remove user id at the time of create and update it after onchange call , is it ok ?